src/peft/tuners/adalora/bnb.py

```diff
@@ -85,37 +74,28 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
 
 if is_bnb_4bit_available():
 
-    class SVDLinear4bit(bnb.nn.Linear4bit, AdaLoraLayer):
+    class SVDLinear4bit(torch.nn.Module, AdaLoraLayer):
         # Low-rank matrix for SVD-based adaptation
         def __init__(
             self,
-            adapter_name,
-            in_features,
-            out_features,
+            base_layer: torch.nn.Module,
```

Comment:
For backward compatibility we should maybe consider keeping the same arg order, ignore the unused kwargs and put the new arguments at the end. This might be too much of an edge case but we never know. What do you think?

Reply:
agreed!